Builtin capabilities fail to start when Keycloak auth is enabled in deployment

## Description

When deploying a builtin capability (e.g., the HTTP capability) in an environment with Keycloak authentication enabled, the capability fails to start because it cannot connect to the Keycloak server. The OIDC client tries to connect to `localhost:8543` instead of using the `AUTH_SERVER` environment variable that is correctly set on the pod.

This appears to be a regression introduced by the recent noauth profile changes (f300eb9, 6e19fc5, 2caf3258, 68405c3, e89a6c25) which added support for authentication-less Wanaku. These changes likely broke the configuration path where capabilities pick up the correct OIDC/Keycloak server URL from the environment.

## Steps to Reproduce

1. Deploy Wanaku with Keycloak authentication enabled
2. Deploy a builtin capability (e.g., http)
3. Observe the capability pod failing to start

## Expected Behavior

The capability should read the `AUTH_SERVER` environment variable and connect to the configured Keycloak instance.

## Actual Behavior

The capability ignores the `AUTH_SERVER` environment variable and tries to connect to the default `localhost:8543`, resulting in a `Connection refused` error and application startup failure.

## Logs

```
INFO exec -a "java" java -XX:MaxRAMPercentage=80.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:+ExitOnOutOfMemoryError -Djava.util.logging.manager=org.jboss.logmanager.LogManager -cp "." -jar /home/jboss/quarkus-run.jar
INFO running in /home/jboss
2026-04-27 09:16:43,814 WARN  [io.qua.oid.com.run.OidcCommonUtils] (vert.x-eventloop-thread-1) OIDC Server is not available:: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8543
Caused by: java.net.ConnectException: Connection refused
    at java.base/sun.nio.ch.Net.pollConnect(Native Method)
    at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:682)
    at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
    at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
    at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:784)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:1583)

2026-04-27 09:16:43,857 ERROR [io.qua.run.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
    at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
    at io.quarkus.runtime.Application.start(Application.java:101)
    at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:144)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:69)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
Caused by: io.smallrye.mutiny.CompositeException: Multiple exceptions caught:
    [Exception 0] java.lang.RuntimeException: OIDC Server is not available
    [Exception 1] io.quarkus.oidc.client.OidcClientException: OIDC server is not available at the 'http://localhost:8543/realms/wanaku' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
    at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureMap$UniOnItemOrFailureMapProcessor.onFailure(UniOnItemOrFailureMap.java:59)
    at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
    at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:64)
    at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap$UniOnFailureFlatMapProcessor.dispatch(UniOnFailureFlatMap.java:85)
    at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap$UniOnFailureFlatMapProcessor.onFailure(UniOnFailureFlatMap.java:60)
    at io.smallrye.mutiny.operators.uni.builders.UniCreateFromPublisher$PublisherSubscriber.onError(UniCreateFromPublisher.java:83)
    at io.smallrye.mutiny.subscription.MultiSubscriberAdapter.onFailure(MultiSubscriberAdapter.java:32)
    at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
    at io.smallrye.mutiny.subscription.SerializedSubscriber.onFailure(SerializedSubscriber.java:101)
    at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.whenFailure(MultiRetryWhenOp.java:165)
    at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$TriggerSubscriber.onError(MultiRetryWhenOp.java:193)
    at io.smallrye.mutiny.subscription.MultiSubscriberAdapter.onFailure(MultiSubscriberAdapter.java:32)
    at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.innerOnFailure(MultiConcatMapOp.java:167)
    at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber$InnerSubscriber.onFailure(MultiConcatMapOp.java:307)
    at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
    at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:104)
    at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownFailure$KnownFailureSubscription.forward(UniCreateFromKnownFailure.java:38)
    at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownFailure.subscribe(UniCreateFromKnownFailure.java:23)
    at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
    at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.request(UniToMultiPublisher.java:74)
    at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.innerOnSubscribe(MultiConcatMapOp.java:109)
    at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber$InnerSubscriber.onSubscribe(MultiConcatMapOp.java:297)
    at io.smallrye.mutiny.converters.uni.UniToMultiPublisher.subscribe(UniToMultiPublisher.java:25)
    at io.smallrye.mutiny.groups.MultiCreate$1.subscribe(MultiCreate.java:165)
    at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.onItem(MultiConcatMapOp.java:119)
    at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
    at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:108)
    at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:139)
    at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:205)
    at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:104)
    at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
    at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
    at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:127)
    at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
    at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:104)
    at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:55)
    at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:37)
    at io.smallrye.mutiny.vertx.DelegatingHandler.handle(DelegatingHandler.java:25)
    at io.vertx.ext.web.client.impl.HttpContext.handleFailure(HttpContext.java:399)
    at io.vertx.ext.web.client.impl.HttpContext.execute(HttpContext.java:388)
    at io.vertx.ext.web.client.impl.HttpContext.next(HttpContext.java:363)
    at io.vertx.ext.web.client.impl.HttpContext.fire(HttpContext.java:330)
    at io.vertx.ext.web.client.impl.HttpContext.fail(HttpContext.java:311)
    at io.vertx.ext.web.client.impl.HttpContext.lambda$handleCreateRequest$5(HttpContext.java:496)
    at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
    at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
    at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
    at io.vertx.core.http.impl.HttpClientImpl.lambda$doRequest$4(HttpClientImpl.java:398)
    at io.vertx.core.net.impl.pool.Endpoint.lambda$getConnection$0(Endpoint.java:52)
    at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:162)
    at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint$Request.handle(SharedClientHttpStreamEndpoint.java:123)
    at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
    at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:335)
    at io.vertx.core.net.impl.pool.SimpleConnectionPool$ConnectFailed$1.run(SimpleConnectionPool.java:380)
    at io.vertx.core.net.impl.pool.Task.runNextTasks(Task.java:43)
    at io.vertx.core.net.impl.pool.CombinerExecutor.submit(CombinerExecutor.java:91)
    at io.vertx.core.net.impl.pool.SimpleConnectionPool.execute(SimpleConnectionPool.java:244)
    at io.vertx.core.net.impl.pool.SimpleConnectionPool.lambda$connect$2(SimpleConnectionPool.java:258)
    at io.vertx.core.http.impl.SharedClientHttpStreamEndpoint.lambda$connect$2(SharedClientHttpStreamEndpoint.java:104)
    at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
    at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
    at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
    at io.vertx.core.impl.future.Composition$1.onFailure(Composition.java:66)
    at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
    at io.vertx.core.impl.future.FailedFuture.addListener(FailedFuture.java:98)
    at io.vertx.core.impl.future.Composition.onFailure(Composition.java:55)
    at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
    at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
    at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
    at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:335)
    at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:351)
    at io.vertx.core.net.impl.NetClientImpl.lambda$connectInternal2$6(NetClientImpl.java:323)
    ...
    Suppressed: io.quarkus.oidc.client.OidcClientException: OIDC server is not available at the 'http://localhost:8543/realms/wanaku' URL. Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example: 'https://localhost:8180/auth/realms/quarkus'
        at io.quarkus.oidc.client.runtime.OidcClientRecorder.toOidcClientException(OidcClientRecorder.java:228)
        at io.quarkus.oidc.client.runtime.OidcClientRecorder$2.apply(OidcClientRecorder.java:138)
        at io.quarkus.oidc.client.runtime.OidcClientRecorder$2.apply(OidcClientRecorder.java:133)
        at io.smallrye.context.impl.wrappers.SlowContextualBiFunction.apply(SlowContextualBiFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureMap$UniOnItemOrFailureMapProcessor.onFailure(UniOnItemOrFailureMap.java:54)
        ... 95 more
    Caused by: java.lang.RuntimeException: OIDC Server is not available
        at io.quarkus.oidc.common.runtime.OidcCommonUtils.lambda$discoverMetadata$4(OidcCommonUtils.java:566)
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
        at io.smallrye.mutiny.operators.uni.UniOnFailureTransform$UniOnFailureTransformProcessor.onFailure(UniOnFailureTransform.java:54)
        ... 93 more
Caused by: [CIRCULAR REFERENCE: java.lang.RuntimeException: OIDC Server is not available]
```

## Additional Context

The `AUTH_SERVER` env var is correctly set on the pod:
```
AUTH_SERVER: http://keycloak-address-in-the.cloud
```

The issue was likely introduced by the noauth profile commits which may have changed how the OIDC configuration properties are resolved or overridden.

--- a/apps/wanaku-router-backend/src/main/resources/application.properties
+++ b/apps/wanaku-router-backend/src/main/resources/application.properties
@@ -44,7 +44,7 @@ quarkus.mcp.server.client-logging.default-level=debug
 quarkus.oidc.enabled=true
 
 # Address of the Keycloak authentication server - adjust to your Keycloak instance
-auth.server=http://localhost:8543
+auth.server=${AUTH_SERVER:http://localhost:8543}
 # Address used by the OIDC proxy -
 auth.proxy=http://localhost:${quarkus.http.port}
 

--- a/capabilities-quarkus-sdk/wanaku-capabilities-base/src/main/resources/application.properties
+++ b/capabilities-quarkus-sdk/wanaku-capabilities-base/src/main/resources/application.properties
@@ -3,7 +3,7 @@ quarkus.qute.strict-rendering=false
 quarkus.oidc-client.enabled=true
 
 # Address of the Keycloak authentication server - adjust to your Keycloak instance
-auth.server=http://localhost:8543
+auth.server=${AUTH_SERVER:http://localhost:8543}
 
 # Address of the KeyCloak authentication server
 quarkus.oidc-client.auth-server-url=${auth.server}/realms/wanaku

